src/components/icon-button/icon-button.test.tsx

```diff
@@ -0,0 +1,153 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+
+import Icon from "../icon";
+import IconButton from ".";
+import { testStyledSystemSpacing } from "../../__spec_helper__/__internal__/test-utils";
+import Logger from "../../__internal__/utils/logger";
+
+jest.mock("../../__internal__/utils/logger");
+
+test("should render with an `Icon` child", () => {
+  render(
+    <IconButton aria-label="icon-button">
+      <Icon type="bin" />
+    </IconButton>
+  );
+
+  const iconButton = screen.getByRole("button", { name: "icon-button" });
+  const icon = screen.getByTestId("icon");
```

Comment:
comment (non-blocking): you could use `within` here and batch 2 of the assertions into 1 if you want but it's fine how it is.

```
  const icon = within(iconButton).getByTestId("icon");
  
   expect(iconButton).toBeVisible();
   expect(icon).toBeVisible();